### PR TITLE
ensure account of native token keeps alive after charge fee

### PR DESCRIPTION
**Pull request by wangjj9219**
charge fee maybe fail even if the account has sufficient tokens for these reasons:

1. https://github.com/AcalaNetwork/Acala/blob/ae1dba93636e7536ca8f49cfc80539460383751a/modules/transaction-payment/src/lib.rs#L676 the account must keep alive after charge fee.

2. https://github.com/AcalaNetwork/Acala/blob/ae1dba93636e7536ca8f49cfc80539460383751a/modules/transaction-payment/src/lib.rs#L593  The `balances::transfer` will executed when swap native token and `transfer` requires the receiver is alive, otherwise the `swap` will fail.

So fix `ensure_can_charge_fee` to guarantee account will keep alive.

### CODE REVIEW

**Review by xlc — APPROVED** [review]
better to have some tests

File: modules/transaction-payment/src/lib.rs [hunk 1]
```diff
@@ -567,17 +567,24 @@ where
...
-				let native_is_enough =
-					<T as Config>::Currency::free_balance(who)
-						.checked_sub(&fee)
-						.map_or(false, |new_free_balance| {
-							<T as Config>::Currency::ensure_can_withdraw(who, fee, reason, new_free_balance).is_ok()
-						});
+				let native_is_enough = <T as Config>::Currency::free_balance(who)
+					.checked_sub(&fee_and_alive_gap)
+					.map_or(false, |new_free_balance| {
+						<T as Config>::Currency::ensure_can_withdraw(who, fee, reason, new_free_balance).is_ok()
```
Comment (zjb0807) [inline modules/transaction-payment/src/lib.rs:586]:
```suggestion
						<T as Config>::Currency::ensure_can_withdraw(who, fee, reason, fee_and_alive_gap).is_ok()
```

Comment (zjb0807) [reply modules/transaction-payment/src/lib.rs:586]: @xlc The old code seems to be wrong.

Comment (xlc) [reply modules/transaction-payment/src/lib.rs:586]: https://github.com/paritytech/substrate/blob/993907f9deb180795b291ba752b9c204165d74dc/frame/support/src/traits/tokens/currency.rs#L103

the last parameter is expected new balance after withdraw

Comment (zjb0807) [reply modules/transaction-payment/src/lib.rs:586]: Ok, It's different from tokens.
https://github.com/open-web3-stack/open-runtime-module-library/blob/3d64077e5ac5539bde336a794308d85f6f5f2b82/tokens/src/lib.rs#L642

Comment (xlc) [reply modules/transaction-payment/src/lib.rs:586]: Yes we removed the last parameter

**Pull request merged**